version: '3.8'

services:
  app-backend:
    container_name: app-backend
    build:
      context: ./backend
      dockerfile: Dockerfile
    ports:
      - "3333:3333"
    environment:
      - FLASK_APP=run.py
      - FLASK_ENV=development
    volumes:
      - ./backend:/app
      - sqlite_data:/app/instance
    networks:
      - app-network
      - simulation

  app-solver:
    container_name: app-solver
    build:
      context: ./solver
      dockerfile: Dockerfile
    ports:
      - "5000:5000"
    volumes:
      - ./solver:/app
    environment:
      - PYTHONUNBUFFERED=1
    networks:
      - app-network

  app-grafana:
    container_name: app-grafana
    image: grafana/grafana:latest
    user: root
    ports:
      - "3001:3000"
    volumes:
      - grafana_data:/var/lib/grafana
      - sqlite_data:/var/lib/grafana/hackathon:ro
    environment:
      - GF_SECURITY_ADMIN_USER=admin
      - GF_SECURITY_ADMIN_PASSWORD=admin
      - GF_SECURITY_ALLOW_EMBEDDING=true
      - GF_AUTH_ANONYMOUS_ENABLED=true
      - GF_PANELS_DISABLE_SANITIZE_HTML=true
    networks:
      - app-network

networks:
  app-network:
    driver: bridge
  simulation:
    name: hackatum24_default
    external: true

volumes:
  grafana_data:
  sqlite_data:
    name: sqlite_data_clean
